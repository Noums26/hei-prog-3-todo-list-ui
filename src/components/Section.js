import React from "react";

const Section = ({ children, title }) => {
  return (
    <div className="todo-flex">
      <h3>{title}</h3>
      <hr />
      {children}
    </div>
  );
};

export default Section;
