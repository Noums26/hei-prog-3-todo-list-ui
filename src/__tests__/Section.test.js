import { create } from "react-test-renderer";
import Section from "../components/Section";

test("should test empty todo section snapshot", () => {
  const section = create(<Section title="TODO" />).toJSON();

  expect(section).toMatchSnapshot();
});

test("should test empty done section snapshot", () => {
  const section = create(<Section title="DONE" />).toJSON();

  expect(section).toMatchSnapshot();
});

test("should test todo section snapshot", () => {
  const todo_section = create(
    <Section title="TODO">
      <div>Tepr</div>
    </Section>
  );

  const ins = todo_section.root.children[0].props.children[2].props.children;

  expect(ins).toBe("Tepr");
});

test("should test done section snapshot", () => {
  const done_section = create(
    <Section title="DONE">
      <div>Tepr</div>
    </Section>
  );

  const ins = done_section.root.children[0].props.children[2].props.children;

  expect(ins).toBe("Tepr");
});
