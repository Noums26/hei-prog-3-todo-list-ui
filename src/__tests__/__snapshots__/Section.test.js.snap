// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test empty done section snapshot 1`] = `
<div
  className="todo-flex"
>
  <h3>
    DONE
  </h3>
  <hr />
</div>
`;

exports[`should test empty todo section snapshot 1`] = `
<div
  className="todo-flex"
>
  <h3>
    TODO
  </h3>
  <hr />
</div>
`;
